ingestion/src/metadata/profiler/profiler/interface/pandas/pandas_profiler_interface.py

```diff
@@ -161,12 +172,8 @@ def _(
         try:
             row = []
             for metric in metrics:
-                for df in dfs:
-                    row.append(
-                        metric(column).df_fn(
-                            df.astype(object).where(pd.notnull(df), None)
-                        )
-                    )
+                metric_resp = metric(column).df_fn(self.dfs)
+                row.append(None if pd.isnull(metric_resp) else metric_resp)
             row_dict = {}
```

Comment:
I think you should populate that dictionnary in your loop above (vs working with index). We should probably rename the variable to `metrics`